File: mobile/android/core/auth/src/main/kotlin/com/slopssaloon/omen/core/auth/AuthMechanism.kt
package com.slopssaloon.omen.core.auth

/**
 * The Omen account sign-in mechanisms (M0c §2.1, M0a §4.2).
 *
 * Provider *connection* (Yahoo/Sleeper/ESPN) is deliberately NOT modeled here — that is a
 * separate concern gated behind M0-BE and out of M3-A scope.
 */
enum class AuthMechanism {
    /** Android Credential Manager → Google ID token → Supabase `signInWithIdToken`. No browser. */
    GOOGLE_ID_TOKEN,

    /** 6-digit email one-time code (not a magic link — avoids mobile deep-link fragility). */
    EMAIL_OTP,

    /** iOS Sign in with Apple. Declared for contract parity; not implemented on Android. */
    APPLE_ID_TOKEN,
}

File: mobile/ios/OmenIOS/OmenIOS/Core/Auth/AuthMechanism.swift
import Foundation

/// Mirrors Android `core/auth/AuthMechanism.kt`. iOS's native ID-token mechanism is Apple, not
/// Google (App Store 4.8: Sign in with Apple is required whenever any third-party/social login
/// is offered). `.googleIDToken` is kept only for contract parity with the Android enum and is
/// not implemented on iOS.
enum AuthMechanism {
    case appleIDToken
    case emailOTP
    case googleIDToken
}
